Full refresh

mareklibra opened the pull request:
User can invoke 'Refresh' to fully reload data

mareklibra committed:
Remove missing VMs

Useful for all-VM-refresh to remove VMs which are not received
from API anymore (means deleted in the meantime).

mareklibra committed:
CLEAR_VM_DISKS action added

mareklibra committed:
Action for shallow VMs refresh

matobet commented on src/reducers/vms.js:
```diff
@@ -27,6 +27,8 @@ function removeMissingVms ({ state, payload: { vmIdsToPreserve } }) {
 
 function updateVmDisk ({ state, payload: { vmId, disk } }) {
+  // TODO: If disk management is implemented then maintain disks separately
+  // and manage references (like for icons). Recently it makes no sense
```
s/Recently/For now :-)

matobet commented on src/reducers/vms.js:
```diff
@@ -35,6 +37,15 @@ function updateVmDisk ({ state, payload: { vmId, disk } }) {
   return state
 }
 
+function clearVmDisks ({ state, payload: { vmId } }) {
```
just a question: why do we need to clear a VM's disks?

matobet commented on src/reducers/vms.js:
```diff
@@ -15,6 +15,17 @@ function removeVms ({ state, payload: { vmIds } }) {
+  const vmIdsToBeDeleted = state.get('vms')
+    .keySeq()
+    .filter(vmId => !vmIdsToPreserve.some(toPreserveId => toPreserveId === vmId))
+    .map(vmId => vmId)
```
the last mapping appears to be a redundant (identity) operation

mareklibra replied:
Right, left over ...

mareklibra committed:
VM disks are set in one operation

due to:
- effectivity
- avoiding 'blick' when rendering
- reuse of code when for single-VM refresh

mareklibra committed:
Cleanup

mareklibra commented:
Done

matobet commented on src/reducers/vms.js:
```diff
@@ -15,9 +15,19 @@ function removeVms ({ state, payload: { vmIds } }) {
+function removeMissingVms ({ state, payload: { vmIdsToPreserve } }) {
+  const vmIdsToBeDeleted = state.get('vms')
+    .keySeq()
+    .filter(vmId => !vmIdsToPreserve.some(toPreserveId => toPreserveId === vmId))
```
please consider using a `Set` instead of doing nested linear lookups to check for presence

mareklibra committed:
REMOVE_MISSING_VMS: use Set instead of Array

Optimization.
